src/inference/include/openvino/runtime/intel_gna/gna_properties.hpp

```diff
@@ -0,0 +1,213 @@
+ * @details For multiple input case, individual scale factors can be passed, using
+ *  map where key is layer name and value is scale factor
+ */
+static constexpr Property<std::map<std::string, float>> scale_factors_per_input{"GNA_SCALE_FACTOR_PER_INPUT"};
+
+/**
+ * @brief if turned on, dump GNA firmware model into specified file
+ */
+static constexpr Property<std::string> firmware_model_image_path{"GNA_FIRMWARE_MODEL_IMAGE"};
+
+/**
+ * @brief Enum to define software acceleration mode
+ */
+enum class ExecutionMode {
+    AUTO = 0,              // Uses Intel GNA if available, otherwise uses software execution mode on CPU.
+    HW = 1,                // Uses Intel GNA if available, otherwise raises an error.
+    HW_WITH_SW_FBACK = 2,  // Uses Intel GNA if available, otherwise raises an error.
+                           // If the hardware queue is not empty, automatically falls back to CPU in the bit-exact mode.
+    SW_EXACT = 3,          // Executes the GNA-compiled graph on CPU performing calculations
+                           // in the same precision as the Intel GNA in the bit-exact mode.
+    SW_FP32 = 4,           // Executes the GNA-compiled graph on CPU but substitutes parameters and calculations
+                           // from low precision to floating point
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const ExecutionMode& execution_mode) {
+    switch (execution_mode) {
+    case ExecutionMode::AUTO:
+        return os << "GNA_AUTO";
+    case ExecutionMode::HW:
+        return os << "GNA_HW";
+    case ExecutionMode::HW_WITH_SW_FBACK:
+        return os << "GNA_HW_WITH_SW_FBACK";
+    case ExecutionMode::SW_EXACT:
+        return os << "GNA_SW_EXACT";
+    case ExecutionMode::SW_FP32:
+        return os << "GNA_SW_FP32";
+    default:
+        throw ov::Exception{"Unsupported execution mode!"};
+    }
+}
+
+inline std::istream& operator>>(std::istream& is, ExecutionMode& execution_mode) {
+    std::string str;
+    is >> str;
+    if (str == "GNA_AUTO") {
+        execution_mode = ExecutionMode::AUTO;
+    } else if (str == "GNA_HW") {
+        execution_mode = ExecutionMode::HW;
+    } else if (str == "GNA_HW_WITH_SW_FBACK") {
+        execution_mode = ExecutionMode::HW_WITH_SW_FBACK;
+    } else if (str == "GNA_SW_EXACT") {
+        execution_mode = ExecutionMode::SW_EXACT;
+    } else if (str == "GNA_SW_FP32") {
+        execution_mode = ExecutionMode::SW_FP32;
+    } else {
+        throw ov::Exception{"Unsupported execution mode: " + str};
+    }
+    return is;
+}
+/** @endcond */
+
+/**
+ * @brief Enum to define compile and execution targets
+ */
+enum class TargetDevice {
+    UNKNOWN = 0,
+    GNA_2_0 = 1,
+    GNA_3_0 = 2,
+};
+
+/** @cond INTERNAL */
+inline std::ostream& operator<<(std::ostream& os, const TargetDevice& target_device) {
+    switch (target_device) {
+    case TargetDevice::UNKNOWN:
+        return os << "";
```

Comment:
@mryzhov @ilya-lavrenov Currently default value is an empty string for this parameter. There is no way to determine what should I return in get_config by default: empty string or `UNKNOWN` because the key is the same for both APIs. I think changing the default value is not safe, because some customers can use this value in their code.